src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -212,5 +230,22 @@ private IEnumerable<IConfigurationSection> GetFeatureDefinitionSections()
                 return _configuration.GetChildren();
             }
         }
+
+        private T ParseFeatureDefinitionSectionEnum<T>(IConfigurationSection configurationSection, string keyword, T enumValue)
+            where T : struct, Enum
+        {
+            string rawValue = configurationSection[keyword];
+
+            //
+            // If the enum is specified, parse it and set the return value
+            if (!string.IsNullOrEmpty(rawValue) && !Enum.TryParse(rawValue, ignoreCase: true, out enumValue))
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.InvalidConfigurationSetting,
+                    $"Invalid {typeof(T)?.Name} with value '{rawValue}' for feature '{configurationSection.Key}'.");
+            }
+
+            return enumValue;
```

Comment:
That doesn't look right in the case where value is empty string. It'll overwrite the value of the enum at the caller.

Reply:
Not sure I understand, it would return the `enumValue` param passed in since it skips the `Enum.TryParse()`. Is it the way we pass in the variable we're setting?